week_app/views.py

```diff
@@ -0,0 +1,225 @@
+        request,  # uzywam funcji render
+        'week_app/died_list.html',
+        {
+            'died': Died.objects.filter(week_number=week_nr),
+            'week_nr': week_nr,
+
+        }
+    )
+
+
+# R (szczegół) z CRUD
+def died_detail_view(request, died_id):  # Widok szczegółów diety
+    died = get_object_or_404(Died, id=died_id)  # Otrzymaj diete z id(baza danych) albo 404
+    # comments = Comment.objects.filter(died=died)
+    # image = Image.objects.filter( died_id=died_id)
+    return render(
+        request,
+        'week_app/died_detail.html',
+        {
+            'died': died,
+            # 'comments': comments,
+            # 'image': image
+
+        }
+    )
+
+
+# U z CRUD
+
+@login_required
+def died_update_view(request, died_id):  # Widok edycji diety
+    died = get_object_or_404(Died, id=died_id)  # Otrzymaj diete z id(baza danych) albo 404
+    if request.method == 'POST':
+        new_died = request.POST.get('died')  # Jeżeli Post to nowa dieta
+        if new_died is not None:  # dodaj nową id i nazwe diety
+            died.name = new_died
+            died.modify_date = timezone.now()  # utwórz date modyfikacji diety
+            died.save()  # save do database
+
+        # image = request.FILES['image']
+        image = request.FILES.get('image')
+        if image is not None:
+            died.image = image
+            died.save()
+
+        form = DiedForm(request.POST, instance=died)  # Tworzę formularz powiązany z modelem diety celem edycji tekstu
+        if form.is_valid():
+            form.save()
+
+        return redirect(
+            'week_app:died_detail',
+            died_id=died_id
+        )
+
+    else:
+        form = DiedForm(instance=died)
+
+    return render(request,
+                  'week_app/died_edit.html',
+
+                  {
+                      'died': died,
+                      'form': form,
+
+                  }
+                  )
+
+
+# D z CRUD
+@login_required
+def died_delete_view(request, died_id, week_nr):
+    died = get_object_or_404(Died, id=died_id)
+
+    if request.method == "GET":
+        return render(
+            request,
+            'week_app/died_confirm_delete.html',
+            {
+                'died': died,
+                'week_nr': week_nr
+
+            }
+        )
+
+    elif request.method == "POST":
+        data = request.POST
+        if 'yes' in data:
+            died.delete()
+
+        return redirect(
+            'week_app:died_list',
+            week_nr=week_nr
+        )
+
+
+""""------------------------------------- Comments----------------------------"""
+@login_required
+def comment_add_view(request, died_id):
+    died = get_object_or_404(Died, id=died_id)
+
+    if request.method == 'POST':
+        form = CommentForm(request.POST)
+        if form.is_valid():
+            comment = form.save(commit=False)
+            comment.died = died
+            comment.created_date = timezone.now()
```

Comment:
tutaj też jak uwzględnisz parametr auto_now_add=True w modelu to ta linijka nie będzie potrzebna